src/main/java/net/bytzo/sessility/mixins/PlayerMixin.java

```diff
@@ -0,0 +1,132 @@
+	 *	 deaths, picked_up, killed_by, leave_game, play_time, total_world_time, time_since_death,
+	 *	 time_since_rest, raid_trigger, raid_win
+	 *
+	 * The Following advancements are probably not useful, or are already covered by other cases:
+	 *	 broken, picked_up, dropped, killed, damage_dealt, mob_kills, player_kills, sneak_time, target_hit
+	 *
+	 * Any of the advancements named ???_one_cm (walk_one_cm, fly_one_cm, etc) *could* be used to track when a player moves,
+	 * except that the events are fired when the player is pushed by the environment making determining actual input difficult.
+	 *
+	 * Final note, surprisingly, damage_dealt is not triggered by the thorns enchantment.
+ */
+
+	/* Advancement action list.  Actions taken within the world that are deliberate and visible. */
+	private static Set<String> advancementActionList =
+		Stream.of("mined", "used", "jump", "drop", "damage_blocked_by_shield", "fish_caught", "sleep_in_bed")
+		.collect(Collectors.collectingAndThen(Collectors.toCollection(HashSet::new),Collections::unmodifiableSet));
+
+	/* Advancement interaction list.	Actions relating to interacting with the environment.
+	 * Wildcard rules apply to advancements starting with "interact_with_", "inspect_", "and open_",
+	 * so they are unlisted here. */
+	private static Set<String> advancementInteractList =
+		Stream.of("crafted", "talked_to_villager", "traded_with_villager", "eat_cake_slice", "fill_cauldron", "use_cauldron",
+							"play_noteblock", "tune_noteblock", "pot_flower", "trigger_trapped_chest", "enchant_item",
+							"play_record", "bell_ring")
+		.collect(Collectors.collectingAndThen(Collectors.toCollection(HashSet::new),Collections::unmodifiableSet));
+
+	/* optionally prints advancement debugging information */
+	private void advancement_debug(String state, String playername, String advancement) {
+		if (!Sessility.settings().properties().advancementDebugMessages) return;
+		LogManager.getLogger().info("{}: {} {}", playername, state, advancement);
+	}
+
+	/* use advancements as potential motility detectors */
+	private void processAdvancement(ServerPlayer player, String advancement, int count) {
+
+		// process advancement action list
+		if (Sessility.settings().properties().advancementActionDetection && advancementActionList.contains(advancement)) {
+			advancement_debug("[action]", player.getName().getString(), advancement);
+			player.resetLastActionTime();
+			return;
+		}
+
+		// process advancement interact list and wildcard filters
+		if (Sessility.settings().properties().advancementInteractDetection &&
+				 (advancementInteractList.contains(advancement) ||
+					advancement.startsWith("interact_with_") ||
+					advancement.startsWith("inspect_") ||
+					advancement.startsWith("open_")
+				 )
+			 ) {
+			advancement_debug("[interact]", player.getName().getString(), advancement);
+			player.resetLastActionTime();
```

Comment:
Similar to the above, we might not want to call `resetLastActionTime()` at it will affect the behavior of the vanilla server idle kick. Instead, we could use `setSessile(true)`.